src/literal.h

```diff
@@ -444,7 +444,27 @@ class Literal {
   Literal avgrUInt(const Literal& other) const;
 };
 
-using Literals = SmallVector<Literal, 1>;
+class Literals : SmallVector<Literal, 1> {
+public:
+  Literals() = default;
+  Literals(std::initializer_list<Literal> init)
+    : SmallVector<Literal, 1>(init) {
+#ifndef NDEBUG
+    for (auto& lit : init) {
+      assert(lit.isConcrete());
```

Comment:
a single unreachable should be ok though?

Reply:
No, I don't think we support unreachable `Literal`s. A single `none` should not be ok either, since that would more correctly be modeled by an empty `Literals`. The fact that none-typed `Literal`s are allowed at all seems like a workaround for the absence of an ergonomic `Option` type in C++ anyway.

Reply:
Sounds good.